crates/biome_js_analyze/src/analyzers/complexity/use_optional_chain.rs

```diff
@@ -531,47 +590,132 @@ impl LogicalAndChain {
                     .ok()?;
             match self.cmp_chain(&branch).ok()? {
                 LogicalAndChainOrdering::SubChain => {
-                    // Here we reduce our main `JsAnyExpression` buffer by splitting the main buffer.
+                    // If the previous branch had other expressions that already
+                    // included the optional chaining operators, we need to
+                    // include them as well.
+                    if let Some(mut prev_branch) = prev_branch {
```

Comment:
This block provides the real fix, together with a similar looking one a little bit below.